.github/workflows/validatePullRequest.yml

```diff
@@ -0,0 +1,41 @@
+name: Validate Pull Request
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ 'feature/*', 'dev' ,'master' ]
+  pull_request:
+    branches: [ 'feature/*', 'dev' ,'master' ]
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: windows-latest
+    env:
+      solutionName: Microsoft.Graph.Core.sln
+      relativePath: ./src/Microsoft.Graph.Core
+    steps:
+      - uses: actions/checkout@v2.4.0
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+
+      - uses: nuget/setup-nuget@v1
+        with:
+          nuget-version: '5.x'
+
+      - name: Setup VSTest
+        uses: darenm/Setup-VSTest@v1
+
+      - run: ./scripts/UpdateNugetCredentials.ps1 -username "${{ secrets.PUBLISH_GH_USERNAME }}" -apiToken "${{ secrets.PUBLISH_GH_TOKEN }}" -nugetFileAbsolutePath "./nuget.config"
+        shell: pwsh
+
+      - name : Nuget Restore
+        run: msbuild -t:restore
```

Comment:
dotnet restore?

Reply:
Hey @baywet,
Due to a current limitation in the dotnet SDK, dotnet commands currently won't build the project due to the some of the current target frameworks in place. See open issue [here](https://github.com/dotnet/sdk/issues/491) in dotnet repo.
We are also currently tracking this via #227 and hence we have to resort to using msbuild for now.